jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -172,14 +211,19 @@ def combine_lists_to_1d_spectrum(wl, fnu, dfnu, wave_units, flux_units):
     """
     wlallarr = np.array(wl)
     fnuallarr = np.array(fnu)
-    dfnuallarr = np.array(dfnu)
+    if dfnu is not None:
+        dfnuallarr = np.array(dfnu)
```

Comment:
fixed this

Reply:
ok, it looks like this should work now.  It might be easier to read if all these `if dfnu is not None` lines are handled in one block since it doesn't look like you need any elses, and don't use the variables in the case where `dfnu is None`.

Reply:
I still think combining these lines would make it easier to read, but that is subjective, so I'll leave that up to you (just want to make sure this wasn't forgotten in the many threads)

Reply:
done

Reply:
This whole logic is confusing but fixing it here is probably out of scope. In particular, I don't understand why casting `np.array` on "a list of Quantity" is necessary. In fact, when you do that, you lose units (which is probably why you have to manually add it back later, but that defeats the purpose of using Quantity in the first place):

```python
>>> a = [1, 2, 3] * u.AA
>>> b = [4, 5, 6] * u.AA
>>> np.array([a, b])
array([[1., 2., 3.],
       [4., 5., 6.]])
```

Reply:
i just kept it how it was before while moving stuff around. we can fix this another time